library/ssl_tls13_server.c

```diff
@@ -2954,6 +2964,49 @@ static int ssl_tls13_prepare_new_session_ticket(mbedtls_ssl_context *ssl,
     return 0;
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+/* RFC 8446 section 4.2.10
+ *
+ * struct {
+ *     select (Handshake.msg_type) {
+ *         case new_session_ticket:   uint32 max_early_data_size;
+ *         ...
+ *     };
+ * } EarlyDataIndication;
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_write_nst_early_data_ext(mbedtls_ssl_context *ssl,
+                                              unsigned char *buf,
+                                              const unsigned char *end,
+                                              size_t *out_len)
+{
+    unsigned char *p = buf;
+    *out_len = 0;
+
+    if (!mbedtls_ssl_session_ticket_allow_early_data(ssl->session)) {
```

Comment:
```suggestion
```
we just have to check the server config when generating tickets.

Reply:
That has been re-written. 

That flag is set at the beginning of `write_new_session_ticket_body()`